testing/regress/ecl/streamread.ecl

```diff
@@ -0,0 +1,86 @@
+outRecord := RECORD
+    STRING10 name;
+    unsigned1  id;
+END;
+
+streamed dataset(outRecord) doRead(const varstring name) := EMBED(C++ : distributed)
+
+    const char * rows[] = {
+        "Gavin     \x01",
+        "Simon     \x02",
+        "Charlotte \x09",
+        "TheEnd    \x00" };
+
+    class StreamCreator : public IRowStream, public RtlCInterface
+    {
+    public:
+        StreamCreator(ICodeContext * _ctx, IEngineRowAllocator * _resultAllocator) : resultAllocator(_resultAllocator)
+        {
+            idx = 0;
+        }
+        RTLIMPLEMENT_IINTERFACE
+
+        virtual const void * nextRow()
+        {
+            if (idx >= sizeof(rows)/sizeof(*rows))
+                return NULL;
+
+            RtlDynamicRowBuilder builder(resultAllocator);
+            memcpy(builder.getSelf(), rows[idx++], 11);
+            return builder.finalizeRowClear(11);
+        }
+
+        virtual void stop()
+        {
+        }
+
+    private:
+        Linked<IEngineRowAllocator> resultAllocator;
+        unsigned idx;
+    };
+
+    #body
+    unsigned numParts = ctx->getNodes();
```

Comment:
In other languages you would need an ECL wrapper that passed them in as extra parameters.